pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -612,3 +618,15 @@ def configure_slurm_ddp(self):
         # notify user the that slurm is managing tasks
         if self.is_slurm_managing_tasks:
             rank_zero_info("Multi-processing is handled by Slurm.")
+
+    @property
+    def _sharded_training_type(self) -> bool:
+        return isinstance(self.training_type_plugin,
+                          (DDPShardedPlugin, DDPSpawnShardedPlugin
+                           )) or self._distrib_type in (DistributedType.DDP_SHARDED, DistributedType.DDP_SHARDED_SPAWN)
+
+    @property
+    def _full_sharded_training_type(self) -> bool:
+        return isinstance(
+            self.training_type_plugin, FullShardedPlugin
+        ) or self._distrib_type == DistributedType.FULL_SHARDED
```

Comment:
Put this logic in their respective Enums, so the logic becomes

`self.training_type_plugin.is_sharded() or self._distrib_type.is_sharded()`